pkg/ddl/create_table.go

```diff
@@ -964,14 +981,68 @@ func handleTableOptions(options []*ast.TableOption, tbInfo *model.TableInfo) err
 
 			tbInfo.TTLInfo = ttlInfo
 			ttlOptionsHandled = true
+		case ast.TableOptionSoftDelete:
+			softdeleteInfo.Enable = op.BoolValue
+			softdeleteHandled = true
+		case ast.TableOptionSoftDeleteRetention:
+			softdeleteInfo.Retention = op.StrValue
+			softdeleteHandled = true
+		case ast.TableOptionSoftDeleteJobInterval:
+			softdeleteInfo.JobInterval = op.StrValue
+			softdeleteHandled = true
+		case ast.TableOptionSoftDeleteJobEnable:
+			softdeleteInfo.JobEnable = op.BoolValue
+			softdeleteHandled = true
+		case ast.TableOptionActiveActive:
+			isActiveActive = op.BoolValue
 		case ast.TableOptionEngineAttribute:
 			return errors.Trace(dbterror.ErrUnsupportedEngineAttribute)
 		}
 	}
+
+	// Set active-active (from database default or table option)
+	tbInfo.IsActiveActive = isActiveActive
+
+	// Set soft-delete info
+	if softdeleteHandled || softdeleteInfo.Enable {
+		tbInfo.SoftdeleteInfo = softdeleteInfo
+	}
+
 	shardingBits := shardingBits(tbInfo)
 	if tbInfo.PreSplitRegions > shardingBits {
 		tbInfo.PreSplitRegions = shardingBits
 	}
+
+	if err := checkSoftDeleteAndActiveActive(tbInfo, true); err != nil {
+		return errors.Trace(err)
+	}
+
+	// Add extra hidden columns for active-active and soft-delete features
+	if tbInfo.IsActiveActive {
+		// Add _tidb_commit_ts hidden column
+		// virtual column has fixed ID
+		commitTSCol := model.NewExtraCommitTSColInfo()
+		commitTSCol.State = model.StatePublic
+		commitTSCol.Offset = len(tbInfo.Columns)
+		tbInfo.Columns = append(tbInfo.Columns, commitTSCol)
+
+		// Add _tidb_origin_ts hidden column
+		originTSCol := model.NewExtraOriginTSColInfo()
```

Comment:
What happen if user create a normal table, with a column named `_tidb_origin_ts`?
And what happen if user create a soft delete table, but they have a custom column named `_tidb_origin_ts`?